### DISCUSSION TITLE
推出无服务器函数+LC的图床（支持Qexo上传！！）

### DISCUSSION THREAD

**Discussion by Fgaoxing** [Qexo/Qexo#243] [Ideas]
算是提前发出来了，下面说一下，怎么用，目前由于懒，没有上传页面，但是Qexo可以上传。

提示：由于vercel限制，图片不能超过3MB，否则会出问题

**Comment by Fgaoxing** [comment]
第一步数据库https://leancloud.app/ 国际版即可，当然国内版也可以！！

**Comment by Fgaoxing** [comment]
第二部新建库，然后新建img表，复制apiId和apiKey

**Comment by Fgaoxing** [comment]
https://vercel.com/new/clone?s=https%3A%2F%2Fgithub.com%2FFgaoxing%2Fimg&showOptionalTeamCreation=false
部署，部署好，然后前往环境变量页面
appId 数据库的id
appKey 数据库key
Key 你的上传密码
referer 域名正表达式匹配，示例：/((https|http)?:\/\/)[^\s]+yt-blog.top/（后面的yt-blog.top,改成你自己的域名！！）

**Reply by Fgaoxing** [reply]
這裏錯了，/((https|http)?://)[^\s]+yt-blog.top/ 前后沒有斜杠，應該爲((https|http)?://)[^\s]+yt-blog.top，否則正常網站也不好使

**Comment by Fgaoxing** [comment]
然后重新部署，打开Qexo后台
![image](https://user-images.githubusercontent.com/92655031/204202885-7dc72593-d76a-4829-a571-33932d7c10b1.png)
按照图中的方式配置，key为你刚才设定的密码，域名改为部署的域名，其中"anti_theft_link":"true"为开启防盗链，如果不开启，则应该设为false

**Comment by Fgaoxing** [comment]
默认情况下，如果不设置path则为/时间戳.png
图床上传时，请不要设置path

**Comment by Fgaoxing** [comment]
防盗链不是整个域名都开启，而是，对于某张图片的防盗链
例如https://img.yt-blog.top/png/beijing.png 会出现禁止的图片（这是背景图片，有防盗链，访问https://www.yt-blog.top/  可以看到背景，不过网站有点慢，稍等一会）
而https://img.yt-blog.top/png/big_logo.png 则可以访问（这是我的图标，友链也用这个，所以不能防盗链）
开启防盗链，如果referer参数不存在或不匹配则会显示报错，访问者不会有影响，但是下载或者盗链这回报错

**Comment by Fgaoxing** [comment]
提示建议使用韩国节点（vercel设置》无服务器函数），选择韩国首尔服务器，速度比较快，
我在黑龙江测速的
我的网站（使用cf的DNS，没有CDN）：返回用时2.1682724952697754
7不图床（有CDN，DNS快）：用时1.0486812591552734
测试图片（完全一样）：
https://bu.dusays.com/2022/07/07/62c6f4297cf9a.jpg
https://img.yt-blog.top/2022/07/07/62c6f4297cf9a.jpg（已删除）

**Reply by Fgaoxing** [reply]
不信，就問杜老師

**Comment by Fgaoxing** [comment]
理论上可以上传任何文件，但有大小限制

**Comment by Fgaoxing** [comment]
404图片和防盗链图片可以没有，如果想要设置请等待真是发布，不过，实际上已经可以使用了！！

**Reply by Fgaoxing** [reply]
有需求，在本条评论下面留言！！！

**Reply by ning0818** [reply]
bug


![image](https://user-images.githubusercontent.com/117955401/206325592-0872f6cc-ece3-4c4d-8a8e-f5c01030cacd.png)

**Reply by ning0818** [reply]
![image](https://user-images.githubusercontent.com/117955401/206325702-42d06e3f-6e24-44c7-bc99-367f0d16f78f.png)

**Reply by Fgaoxing** [reply]
@ning0818 请截图环境变量
appKey不要截图
其他的要进行截图，然后到部署》本次部署》函数
复制日志到这里

**Reply by ning0818** [reply]
刷新好了

**Comment by Fgaoxing** [comment]
@am-abudu 給個“值得一看”唄

**Comment by Fgaoxing** [comment]
绝对不是bug



在 2022-12-08 08:20:00，"ygblog" ***@***.***> 写道：

bug
